### PR TITLE
tmod4 Cray collections

### PR DESCRIPTION
**Pull request by jgphpc**
Fix for https://github.com/eth-cscs/reframe/issues/1526

Provided that the config is:

```json
        {                                                                       
            'name': 'PrgEnv-cray',                                              
            'modules': [                                                        
                'PrgEnv-cray_20.08',                                            
            ]                                                                   
        },                                                                      
        {                                                                       
            'name': 'PrgEnv-gnu',                                               
            'modules': [                                                        
                'PrgEnv-gnu_20.08',                                             
            ]                                                                   
        },                            
```

and `module savelist` has:

```python
Named collection list:
 1) PrgEnv-cray   2) PrgEnv-cray_20.08   3) PrgEnv-gnu   4) PrgEnv-gnu_20.08  
```

> module show CrayGNU/.20.08

```
/apps/eiger/UES/tools/easybuild/modules/all/CrayGNU/.20.08:
module-whatis   dummy
```

then the generated scripts look good:

```c
module restore PrgEnv-gnu_20.08
module load CrayGNU/.20.08
module load ddt/20.1.1-Suse-15.0
```

### CODE REVIEW

**Comment by jgphpc** [comment]
It is possible to test lmod on dom with: `source /usr/share/lmod/8.3.1/init/bash`

**Comment by jgphpc** [comment]
Note to myself:

`pytest test_modules.py`

- tmod3: VERSION=3.2.11.4
- tmod4: Release 4.6.0
- lmod: version 8.3.1

All 3 tests pass with: `41 passed, 20 skipped`

> tmod4:
```
unalias ml
module rm modules
source /apps/common/UES/reframe/modules/4.6.0/init/bash
```

File: reframe/core/modules.py [hunk 1]
```diff
@@ -659,18 +670,64 @@ def _exec_module_command(self, *args, msg=None):
...
+        # if not namespace['_mlstatus']:
+        #     # _mlstatus is set by the TMod4 Python bindings
+        #     if msg is None:
+        #         msg = 'modules system command failed: '
+        #         if isinstance(completed.args, str):
+        #             msg += completed.args
+        #         else:
+        #             msg += ' '.join(completed.args)
+        #
+        #     raise EnvironError(msg)
```
Comment (vkarak) [inline reframe/core/modules.py:682]: Why did you comment this out?

**Comment by vkarak** [comment]
Replaced by #1586.

**Pull request closed without merging**